provider/hybrid.go

```diff
@@ -65,8 +70,78 @@ func (dp dockerHybridProvider) CheckConfig(ctx context.Context, request *rpc.Che
 }
 
 func (dp dockerHybridProvider) DiffConfig(ctx context.Context, request *rpc.DiffRequest) (*rpc.DiffResponse, error) {
-	// Delegate to the bridged provider, as native Provider does not implement it.
-	return dp.bridgedProvider.DiffConfig(ctx, request)
+	urn := resource.URN(request.GetUrn())
+	label := fmt.Sprintf("%s.DiffConfig(%s)", dp.name, urn)
+	logging.V(9).Infof("%s executing", label)
+
+	var err error
+	request.Olds, err = dp.unwrapJSONConfig(label, request.GetOlds())
+	if err != nil {
+		return nil, fmt.Errorf("error unwrapping old config: %w", err)
+	}
+	request.News, err = dp.unwrapJSONConfig(label, request.GetNews())
+	if err != nil {
+		return nil, fmt.Errorf("error unwrapping new config: %w", err)
+	}
+	request.OldInputs, err = dp.unwrapJSONConfig(label, request.GetOldInputs())
+	if err != nil {
+		return nil, fmt.Errorf("error unwrapping old inputs: %w", err)
+	}
+
+	ignoreChanges := []string{"registryAuth[*].password", "registryAuth[*].username"}
+	ignoreChanges = append(ignoreChanges, request.IgnoreChanges...)
+	request.IgnoreChanges = ignoreChanges
+
+	res, err := dp.bridgedProvider.DiffConfig(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	// if the diff is empty, it means it only contained changes to username and password which we ignored
```

Comment:
Peculiar! What was res.Changes in this case before re-assignment of it to DIFF_NONE?

Reply:
It was diff unknown which would delegate diff’ing back to the engine. The engine detects a diff though because it doesn’t have the logic to unwrap the nested json config.

i feel like this is an edge case in plugin.DiffConfig that’s not properly handled. But I’ll dig deeper